os/drivers/input/ist415.c

```diff
@@ -0,0 +1,331 @@
+			data->point[i].y  = (p_evt_coord->y_11_4 << 4) | p_evt_coord->y_3_0;
+			data->point[i].h = 0;
+			data->point[i].w = 0;
+			data->point[i].pressure = 0;
+			switch (p_evt_coord->tsta) {
+			case 1:
+				data->point[i].flags = TOUCH_DOWN;
+				break;
+			case 2:
+				data->point[i].flags = TOUCH_MOVE;
+				break;
+			case 3:
+				data->point[i].flags = TOUCH_UP;
+				break;
+			}
+			touchvdbg("COORDINATES: id %d status %d type %d x : %d y : %d\n", data->point[i].id, p_evt_coord->tsta, (p_evt_coord->ttype_3_2 << 2) | p_evt_coord->ttype_1_0, data->point[i].x, data->point[i].y);
+		}
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * Name: ist415_enable
+ ****************************************************************************/
+
+static void ist415_enable(struct touchscreen_s *dev)
+{
+	struct ist415_dev_s *priv = (struct ist415_dev_s *)dev->priv;
+	priv->ops->irq_enable();
+}
+
+/****************************************************************************
+ * Name: ist415_disable
+ ****************************************************************************/
+
+static void ist415_disable(struct touchscreen_s *dev)
+{
+	struct ist415_dev_s *priv = (struct ist415_dev_s *)dev->priv;
+	priv->ops->irq_disable();
+}
+
+/****************************************************************************
+ * Name: is_touchSet
+ ****************************************************************************/
+
+static bool ist415_istouchSet(struct touchscreen_s *dev)
+{
+	struct ist415_dev_s *priv = (struct ist415_dev_s *)dev->priv;
+	return priv->int_pending;
+}
+
+/****************************************************************************
+ * Name: ist415_read
+ ****************************************************************************/
+
+static int ist415_read(struct touchscreen_s *dev, FAR char *buffer)
+{
+	irqstate_t flags;
+	struct ist415_dev_s *priv = (struct ist415_dev_s *)dev->priv;
+	if (priv->int_pending) {
+		flags = enter_critical_section();
+		priv->int_pending = false;
+		leave_critical_section(flags);
+		return ist415_get_touch_data(priv, buffer);
+	} else {
+		touchvdbg("No data to read\n");
+	}
+	return OK;
+}
+
+/****************************************************************************
+ * Name: touch_interrupt
+ ****************************************************************************/
+
+#ifdef CONFIG_TOUCH_POLL
+
+static void touch_interrupt(struct ist415_dev_s *priv)
+{
+
+	FAR struct touchscreen_s *upper = priv->upper;
+	irqstate_t state = enter_critical_section();
+	priv->int_pending = true;
+	leave_critical_section(state);
+	if (upper->notify_touch) {
+		upper->notify_touch(upper);
+	}
+}
+
+#elif CONFIG_TOUCH_CALLBACK
+
+static void touch_interrupt(struct ist415_dev_s *priv)
+{
+	FAR struct touchscreen_s *upper = priv->upper;
+	priv->ops->irq_enable();
+	work_queue(HPWORK, &ist415_work, get_touch_data, priv, 0);
```

Comment:
This is kernel side work queue.
if app is dead, work_queue operation might corrupted.
Could you please find other way to notify to application?